substrate/frame/referenda/src/lib.rs

```diff
@@ -249,6 +222,14 @@ pub mod pallet {
 		type BlockNumberProvider: BlockNumberProvider;
 	}
 
+	#[pallet::extra_constants]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		#[pallet::constant_name(Tracks)]
+		fn tracks() -> Vec<Track<TrackIdOf<T, I>, BalanceOf<T, I>, BlockNumberFor<T, I>>> {
+			T::Tracks::tracks().map(|t| t.into_owned()).collect()
+		}
+	}
```

Comment:
But having this as a constant is incompatible with the trait usage.
In the trait we want to make use of storage to have dynamic tracks. But this being a constant means the track must not change.

I think this PR is inconsistent.
Maybe we should not extend referenda and let other people use a different pallet if they need dynamic tracks.

Or we should make it a view function and remove the constant, which is a bigger breaking change.